lib/booking_manager.rb

```diff
@@ -0,0 +1,137 @@
+require 'date'
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'block_room'
+require_relative 'helper_method'
+
+module Hotel
+  class Booking_Manager
+
+    attr_reader :rooms, :hotel_reservations, :block_reservations
+
+    ROOM_NUMBERS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
+    ROOM_COST = 200.0
+
+    def initialize
+      @rooms = load_rooms
+      @block_reservations = []
+    end
+
+### RESRVATION METHODS ###
+    def reserve_room(input)
+      search_all_rooms_availability(input[:check_in_date], input[:check_out_date])
+
+      reserve = { name: input[:name],
+        room_number: input[:room_number],
+        check_in_date: input[:check_in_date],
+        check_out_date: input[:check_out_date]
+      }
+      reservation = Hotel::Reservation.new(reserve)
+      Hotel::Helper_Method.connect_reservation_to_room_and_sort(@rooms, input[:room_number], reservation)
+    end
+
+    def list_reservations(date)
+      reservations = []
+      @rooms.each do |room|
+        index = Hotel::Helper_Method.binary_search_reservations_return_index_if_found(room.reservations,date)
+        if index == nil
+          next
+        elsif
+          reservations << room.reservations[index]
+        end
```

Comment:
Hm. This `if ... elsif` feels weird, does it work to turn it into `if index != nil`?